lib/vagrant-cloudstack/action/run_instance.rb

```diff
@@ -142,18 +117,18 @@ def call(env)
             options = {
                 :display_name => display_name,
                 :group        => group,
-                :zone_id      => zone_id,
-                :flavor_id    => service_offering_id,
-                :image_id     => template_id
+                :zone_id      => @zone.id,
+                :flavor_id    => @service_offering.id,
+                :image_id     => @template.id
             }
 
-            options['network_ids'] = network_id if !network_id.nil?
-            options['security_group_ids'] = security_group_ids if !security_group_ids.nil?
-            options['project_id'] = project_id if project_id != nil
-            options['key_name']   = keypair if keypair != nil
-            options['name']       = hostname if hostname != nil
-            options['ip_address'] = private_ip_address if private_ip_address != nil
-            options['disk_offering_id'] = disk_offering_id if disk_offering_id != nil
+            options['network_ids'] = [@network.id] unless @network.id.nil?
```

Comment:
This modification, threw me completely off :)

If we set `options['network_ids'] = [@network.id]`, then fog (the cloudstack API client) will throw an error about not being able to verify credentials! So I started to look into how credentials are passed to fog and I just couldn't find anything wrong! 
the `_ids` is indeed confusing. It hints at multiple ids! but it can only take one.